Cabal/Distribution/Simple/GHC.hs

```diff
@@ -1179,12 +1193,14 @@ installLib verbosity lbi targetDir dynlibTargetDir _builtDir _pkg lib clbi = do
     ghciLibName    = Internal.mkGHCiLibName uid
     sharedLibName  = (mkSharedLibName compiler_id) uid
 
-    hasLib    = not $ null (libModules lib)
+    hasLib    = not $ null (allLibModules lib clbi)
                    && null (cSources (libBuildInfo lib))
+    has_code = not (componentIsIndefinite clbi)
+    whenHasCode = when has_code
```

Comment:
This style with `whenHasCode` seems nicer and in keeping with the local style. So we do this here in `installLib`, but if you look at the equivalent changes in `buildOrReplLib` then it's not quite so nice. Can we do it there like we do it here, e.g. adjusting the logic of the `ifReplLib` rather than altering the use sites?